src/tool/subcommands/api_cmd/test_snapshot.rs

```diff
@@ -0,0 +1,158 @@
+// Copyright 2019-2024 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use super::filter::EthEventHandler;
+use crate::{
+    chain::ChainStore,
+    chain_sync::{network_context::SyncNetworkContext, SyncConfig, SyncStage},
+    db::MemoryDB,
+    genesis::{get_network_name_from_genesis, read_genesis_header},
+    libp2p::{NetworkMessage, PeerManager},
+    lotus_json::HasLotusJson,
+    message_pool::{MessagePool, MpoolRpcProvider},
+    networks::ChainConfig,
+    rpc::{RPCState, RpcCallSnapshot, RpcMethod as _, RpcMethodExt as _},
+    shim::address::{CurrentNetwork, Network},
+    state_manager::StateManager,
+    KeyStore, KeyStoreConfig,
+};
+use base64::prelude::*;
+use openrpc_types::ParamStructure;
+use parking_lot::RwLock;
+use std::{path::Path, sync::Arc};
+use tokio::{sync::mpsc, task::JoinSet};
+
+pub async fn run_test_from_snapshot(path: &Path) -> anyhow::Result<()> {
+    CurrentNetwork::set_global(Network::Testnet);
+    let mut run = false;
+    let snapshot_bytes = std::fs::read(path)?;
+    let snapshot_bytes = if let Ok(bytes) = zstd::decode_all(snapshot_bytes.as_slice()) {
+        bytes
+    } else {
+        snapshot_bytes
+    };
+    let snapshot: RpcCallSnapshot = serde_json::from_slice(snapshot_bytes.as_slice())?;
+    let db_bytes = BASE64_STANDARD.decode(&snapshot.db)?;
+    let db = Arc::new(match MemoryDB::deserialize_from(db_bytes.as_slice()) {
```

Comment:
Can we use CAR files rather than serialized memory-dbs?

Reply:
Yes, that's doable. The only downside is that CAR DB is not capable of storing non-content-addressable data(e.g. HEAD -> head_tsk) thus at least 2 files(CAR db, and setting DB+request+desired response) are required for a test case, while with the current solution a single test-snap file is sufficient.